### ISSUE TITLE
Filtering tags

### ISSUE THREAD

**Issue by steeevieee** [skavanagh/EC2Box#20]
I'm new to this but loving what you've done !

I'm struggling with filtering tags at the moment. I want to create profile(s) so that my dev team can access non-production hosts so the widcards are Name=*QA* and Name=*DEV*. 

When filtering in the composite terminal list, either one of those brings up the correct hosts; but I can't use multiple to bring up all the relevant hosts no matter what I try. If I try comma separating the filters (eg Name=*QA*,Name=*PT*) then it only shows the hosts that match the last term

Next up I tried creating individual profiles and applying them to an account. The account can now see only the hosts for the last profile I add, instead of all the hosts the profiles match.

Is there a way to stack up filters ? Maybe a way of adding an OR statement to matching ?

**Comment by skavanagh** [comment]
Looks like it's doing an AND even when it's in separate profiles.  I'll dig into it this week. Thanks!

**Comment by aambekar** [comment]
Hi Sean,
I can help u a bit with this issue. I have fixed it in my branch. The issue is if the tag key is same in different filters like Name=QA and Name=DEV, it gets replaced with the last tag-pair put in the Hash Map.

I used Guava MultiMap to solve this issue. This map basically is Map<String,Collection<String>>
http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Multimap.html

- Ambrish Ambekar
